### PR TITLE
first pass at implementing channels config

### PR DESCRIPTION
**Pull request by jchris**
I just ran this and it doesn't work yet, but I think it is most of the work. I'll give it another pass and then ask if I need help connecting the dots.

this is related to #4

### CODE REVIEW

**Comment by tleyden** [comment]
@jchris any update on this?

Also, do you need to rebase?  It's showing conflicts.

**Comment by tleyden** [comment]
@jchris bump -- can you rebase?

Also, @zgramana mentioned there might need to be some additional headers added

**Comment by zgramana** [comment]
@tleyden is partially misspoke: not a header but an additional query param: `filter=sync_gateway/bychannel`

**Comment by tleyden** [comment]
@zgramana which endpoint are you referring to btw?

@jchris looks like you might need to add that somewhere..

**Comment by jchris** [comment]
I just rebased it (force pushed to the channels branch), not sure why they aren't showing here.

It's not working yet (the URLs don't have the channel param in them). Traun you can probably tell instantly what I didn't see. Lemme know if you need help connecting the dots.

**Comment by jchris** [comment]
scratch that, maybe it is working. at least the tests pass :)

**Comment by tleyden** [comment]
Ready to merge?

**Comment by jchris** [comment]
Actually I just looked into it and the tests are passing but it's not working with my config file so maybe I am parsing it wrong, gonna see if I can figure it out.

**Comment by jchris** [comment]
should be fixed, here is example config

```json
{
    "changes_feed_limit": 50,
    "continuous_retry_time_ms": 15000,
    "replications":{
	"db-local":{
	    "source_url": "http://localhost:4985",
	    "source_db": "retail",
	    "target_db": "regional",
      "channels" : ["reduction"],
	    "lifecycle": "continuous"
	}
    }
}

```

**Comment by tleyden** [comment]
A few requests:

* Remove unnecessary logging
* Rebase -i and squash this into a single commit, then force push

**Comment by jchris** [comment]
done

File: cli/cli_parser.go [hunk 1]
```diff
@@ -36,7 +37,7 @@ func (r ReplicationsConfigJson) Export() (ReplicationsConfig, error) {
 	result.ChangesFeedLimit = r.ChangesFeedLimit
 	result.ContinuousRetryTimeMs = r.ContinuousRetryTimeMs
 	for k, v := range r.ReplicationsMap {
-		logg.LogTo("SYNCTUBE", "k: %v, v: %v", k, v)
+		// logg.LogTo("SYNCTUBE", "k: %v, v: %v", k, v)
```
Comment (jchris) [inline cli/cli_parser.go]: I took this out as long as I was making logs quieter...

File: changes_feed_parameters.go [hunk 2]
```diff
@@ -58,6 +59,10 @@ func (p ChangesFeedParams) String() string {
 	if len(p.SequenceNumber()) > 0 {
 		params = fmt.Sprintf("%v&since=%s", params, p.SequenceNumber())
 	}
-	logg.LogTo("TEST", "changesFeedParams: %v", params)
+	logg.LogTo("SYNCTUBE", "p.channels: %v", p.channels)
```
Comment (tleyden) [inline changes_feed_parameters.go]: I think we should either take this out or rename it to be a little more clear, like "Replication filter channels:"

File: cli/cli_parser.go (hunk 1 above)
Comment (tleyden) [inline cli/cli_parser.go]: Just remove it rather than commenting it ..

File: changes_feed_parameters.go (hunk 2 above)
Comment (jchris) [inline changes_feed_parameters.go]: took it out since we are logging the URL already

**Comment by tleyden** [comment]
Got a build failure .. https://drone.io/github.com/couchbaselabs/sg-replicate/31

**Comment by jchris** [comment]
passing https://drone.io/github.com/couchbaselabs/sg-replicate/33

**Pull request merged**